<launch>
  
  <!-- launch file for processing pointclouds from aubo_control main_LIDAR.launch -->
  <env name="LIBGL_ALWAYS_SOFTWARE" value="1"/>

  <!-- read arg value load the parameters from 'main' config file-->
  <arg name="config" default="aubo_i10_lidar" />
  <param name="param" value="$(arg config)"/> 
  <rosparam file="$(find seam_detection)/config/$(arg config).yaml" /> 
 
  <!-- setup the transformations and frames for the system-->
  <node
    pkg="seam_detection"
    name="seam_detection_tf"
    type="seam_detection_tf"
    output="screen">
  </node>
  
  <!-- get the cloud published by the aubo robot arm -->
  <node                   
    pkg="seam_detection"          
    name="get_cloud"   
    type="get_cloud"
    output="screen">
  </node>

  <!-- save the cloud published by the aubo robot arm --> 
  <node                   
    pkg="seam_detection"          
    name="filter_cloud"   
    type="filter_cloud"
    output="screen">
  </node>

  <!-- processing of lidar data for registration -->
  <node
    pkg="seam_detection"
    name="register_clouds"
    type="register_clouds"
    output="screen">
  </node>


  <!-- start aubo i10 model (this breaks tf) -->
  <arg name="model" default="$(find aubo_description)/urdf/aubo_i10.urdf"/>
  <param name="robot_description" command="$(find xacro)/xacro --inorder $(arg model)" /> 

  <!-- start RVIZ --> 
  <node 
    name="rviz20"
    pkg="rviz"
    type="rviz"
    args="-d $(find seam_detection)/rviz/aubo_i10_lidar.rviz">
  </node>

</launch>

